rockcraft/services/lifecycle.py

```diff
@@ -45,14 +45,25 @@ def setup(self) -> None:
         # This inner import is necessary to resolve a cyclic import
         from rockcraft.services import RockcraftServiceFactory
 
+        # this will be removed once part_has_slices is added to craft-parts
+        def part_has_slices(data: dict[str, Any]) -> bool:
+            stage_packages = data.get("stage-packages", [])
+            return any(name for name in stage_packages if "_" in name)
+
```

Comment:
still within this PR right? Since we're waiting for https://github.com/canonical/craft-parts/pull/637/

Reply:
yes